accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,238 @@
+dotnet
+
+CLI
+
+- Has a means to show outdated packages
+- Has a means to show deprecated/vulnerable packages
+- Has a means to update a single package
+- Does not have means to update multiple packages at once
+
+### Gaps in CLI experience
+
+- No means to update to all latest packages on the CLI
+- No means for update heuristics, such as, update my vulnerable packages
+
+This proposal is focused on bringing the CLI experience up to funcional parity with the experience in Visual Studio and make developers more productive by providing a single command to update all packages to the latest version and a way to update all vulnerable packages.
+
+## User experience
+
+### Updating all packages to the latest version
+
+When users want to ensure their NuGet dependencies are up to date, they can run a command in .NET CLI `dotnet package update` for a project or a solution.
+
+```CLI
+C:\> dotnet package update [<SOLUTION_PATH>|<PROJECT_PATH_>]
+
+Fixing outdated packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Updated UmbracoForms 8.4.1 to 8.7.1
+  Updated Newtonsoft.Json 11.0.1 to 13.0.3
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Updated 3 packages in 36 scanned packages.
+```
+
+### Fixing vulnerabilities
+
+When users run `dotnet build` or `dotnet restore` commands in CLI, if they see any warnings related to vulnerabilities in their project’s NuGet dependencies, they can run `dotnet package update --vulnerable` CLI command to try to remediate all the vulnerabilities. This includes both direct and transitive.
+
+```CLI
+C:\ContosoApp\> dotnet package update --vulnerable
+
+Fixing vulnernable packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Upgrading UmbracoForms 8.4.1 to 8.7.1
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Fixed 2 packages in 36 scanned packages.
+```
+
+In the future this can be extended to fix deprecated versions.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### dotnet package update
+
+The `dotnet package update` command will update all the NuGet dependencies to the latest versions.
+
+```CLI
+C:\> dotnet package update --help
+dotnet package update [<PROJECT>|<SOLUTION>] [--vulnerable] [--mode <MODE>] [-v|--verbosity <LEVEL>] [--json] 
+```
+
+#### --vulnerable
+
+This flag will try to update only packages that have direct or transitive vulnerabilities. The remediation is calculated with an implicit dotnet audit to then apply directly to a resulting package graph. It can add packages, remove packages, and update packages depending on the problem it's attempting to resolve. It does not take into consideration downgrading to a compatible version if a higher one has already been specified.
```

Comment:
```suggestion
This flag will try to update only packages that have direct or transitive vulnerabilities.
The remediation is calculated with an implicit dotnet audit to then apply directly to a resulting package graph.
It can add packages, remove packages, and update packages depending on the problem it's attempting to resolve.
It does not take into consideration downgrading to a compatible version if a higher one has already been specified.
```

I like one sentence per line, because it makes it easier to comment on specific things.

In particular:

1. In another comment, I wrote that if we tag packages promoted to top level, then later on we can remove it as a top level package when we detect it's no longer needed as top level. Without this feature, I'm not sure I can think of any scenarios where we will remove a package reference.
2. I shouldn't be working this late at night, but at the moment, I don't understand the last sentence. Is this in reference to packages where new versions no longer support target frameworks that the current project is using, so while the higher package version no longer has a known vulnerability, it's not compatible with the project? The "if a higher one has already been specified" part also throws me off. At this time of night, I can't guess what this means.